examples_tests/tests.py

```diff
@@ -0,0 +1,250 @@
+                raise
+            else:
+                time.sleep(sleep)
+                timeout -= sleep
+
+
+def _run_command_through_ssh(ip, port, command):
+    ssh_command = ['ssh', 'ubuntu@' + ip, '-p', port]
+    ssh_command.extend(_get_ssh_options())
+    ssh_command.extend(command)
+    return subprocess.check_output(ssh_command).decode("utf-8")
+
+
+def _get_ssh_options():
+    return [
+        '-o', 'UserKnownHostsFile=/dev/null',
+        '-o', 'StrictHostKeyChecking=no',
+        '-i', os.path.join(os.getenv('HOME'), '.ssh', 'id_rsa')
+    ]
+
+
+def _copy_file_through_ssh(ip, port, user, local_file_path, remote_file_path):
+    scp_command = ['scp', '-P', port]
+    scp_command.extend(_get_ssh_options())
+    scp_command.extend([
+        local_file_path, '{}@{}:{}'.format(user, ip, remote_file_path)])
+    subprocess.check_call(scp_command)
+
+
+class TestSnapcraftExamples(testscenarios.TestWithScenarios):
+
+    vm_process = None
+
+    scenarios = [
+        ('downloader-with-wiki-parts', {
+            'dir': 'downloader-with-wiki-parts',
+            'name': 'downloader',
+            'version': '1.0',
+            }),
+        ('godd', {
+            'dir': 'godd',
+            'name': 'godd',
+            'version': '1.0',
+            }),
+        ('gopaste', {
+            'dir': 'gopaste',
+            'name': 'gopaste',
+            'version': '1.0',
+            }),
+        ('java-hello-world', {
+            'dir': 'java-hello-world',
+            'name': 'java-hello-world',
+            'version': '0',
+            }),
+        ('libpipeline', {
+            'dir': 'libpipeline',
+            'name': 'pipelinetest',
+            'version': '1.0',
+            'internal_tests_commands': [
+                ('/apps/bin/pipelinetest.pipelinetest',
+                 'running ls | grep c\n'
+                 'custom libpipeline called\n'
+                 'include\n')],
+            }),
+        ('py2-project', {
+            'dir': 'py2-project',
+            'name': 'spongeshaker',
+            'version': '0',
+            }),
+        ('py3-project', {
+            'dir': 'py3-project',
+            'name': 'spongeshaker',
+            'version': '0',
+            }),
+        ('qmldemo', {
+            'dir': 'qmldemo',
+            'name': 'qmldemo',
+            'version': '1',
+            }),
+        ('ros', {
+            'dir': 'ros',
+            'name': 'ros-example',
+            'version': '1.0',
+            }),
+        ('shout', {
+            'dir': 'shout',
+            'name': 'shout',
+            'version': '0.52.0',
+            }),
+        ('tomcat-maven-webapp', {
+            'dir': 'tomcat-maven-webapp',
+            'name': 'tomcat-webapp-demo',
+            'version': '1.0',
+            }),
+        ('webcam-webui', {
+            'dir': 'webcam-webui',
+            'name': 'webcam-webui',
+            'version': '1',
```

Comment:
maybe add an `internal_tests_commands` to run `snappy config webcam-webui` (but for this to pass, you need to rebase with master for this to pass, so a different commit might be good).